fabric-gamerule-api-v1/src/main/java/net/fabricmc/fabric/mixin/gamerule/client/ScreenAccessor.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.gamerule.client;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Accessor;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.Screen;
+
+@Mixin(Screen.class)
+public interface ScreenAccessor {
+	@Accessor("client")
+	MinecraftClient getClient();
```

Comment:
This shouldn't exist, should just use `MinecraftClient.getInstance()`

Reply:
Though the client is a singleton, I don't think referring to the singleton would be great if mojang next snapshot decides the client is no longer a singleton. Plus in a screen assuming you aren't accessing it from outside, would you use the field/getter or the singleton call

Reply:
There are 212 usages of `MinecraftClient.getInstance()` and that includes things like `AbstractButtonWidget` using that to get the sound manager, a lot more things would break before yours imo.